include/git2/odb.h

```diff
@@ -171,6 +173,40 @@ GIT_EXTERN(int) git_odb_exists(git_odb *db, const git_oid *id);
 GIT_EXTERN(int) git_odb_exists_prefix(
 	git_oid *out, git_odb *db, const git_oid *short_id, size_t len);
 
+typedef struct git_odb_expand_id {
+	/** The object ID to expand */
+	git_oid id;
+
+	/** The length of the object ID */
+	unsigned short length;
+
+	/** The type of the object (optionally) */
+	git_otype type;
```

Comment:
We mark this as optional, which would generally be understood as "leave zeroed and we don't do anything" but leaving this field zeroed means `GIT_OBJ__EXT1`, and won't pass the check against `GIT_OBJECT_ANY` which is what we actually mean by optional here.

It should say to set to `_ANY` if you don't want to enforce the type.